app/src/main/java/com/antosik/benchproject/app/common/BenchBottomNavigationView.kt

```diff
@@ -0,0 +1,53 @@
+package com.antosik.benchproject.app.common
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.animation.TranslateAnimation
+import androidx.core.view.isVisible
+import com.google.android.material.bottomnavigation.BottomNavigationView
+
+class BenchBottomNavigationView(context: Context, attrs: AttributeSet) : BottomNavigationView(context, attrs) {
+
+    private val fromXDelta = 0f
+    private val toXDelta = 0f
+    private val fromYDelta = 0f
+    private val toYDelta = 0f
+
+    fun show() {
+        if (!isVisible) {
+            slideUp()
+        }
+    }
+
+    fun hide() {
+        slideDown()
+    }
+
+    private fun slideUp() {
+        visibility = VISIBLE
+        TranslateAnimation(
+            fromXDelta,
+            toXDelta,
+            height.toFloat(),
+            toYDelta
+        ).let {
+            it.duration = 500
+            it.fillAfter = true
+            startAnimation(it)
+        }
+    }
+
+    private fun slideDown() {
+        TranslateAnimation(
+            fromXDelta,
+            toXDelta,
+            fromYDelta,
+            height.toFloat()
+        ).let {
+            it.duration = 500
+            it.fillAfter = true
+            startAnimation(it)
+        }
+        visibility = GONE
```

Comment:
Why the visibility is set after starting animation here but in case of `slideUp` it is before starting animation?

Reply:
Because I think it will be more clear. But when I changed it works.